core/templates/pages/signup-page/signup-page.component.ts

```diff
@@ -16,213 +16,218 @@
+import { AlertsService } from 'services/alerts.service';
+import { UrlService } from 'services/contextual/url.service';
+import { WindowRef } from 'services/contextual/window-ref.service';
+import { LoaderService } from 'services/loader.service';
+import { SiteAnalyticsService } from 'services/site-analytics.service';
+import { FocusManagerService } from 'services/stateful/focus-manager.service';
+import { LicenseExplanationModalComponent } from './modals/license-explanation-modal.component';
+import { RegistrationSessionExpiredModalComponent } from './modals/registration-session-expired-modal.component';
+import { SignupPageBackendApiService } from './services/signup-page-backend-api.service';
+
+@Component({
+  selector: 'oppia-signup-page',
+  templateUrl: './signup-page.component.html'
+})
+export class SignupPageComponent {
+  MAX_USERNAME_LENGTH = AppConstants.MAX_USERNAME_LENGTH;
+  warningI18nCode = '';
+  siteName = AppConstants.SITE_NAME;
+  submissionInProcess = false;
+  usernameCheckIsInProgress = false;
+  showEmailSignupLink = false;
+  emailSignupLink = AppConstants.BULK_EMAIL_SERVICE_SIGNUP_URL;
+  username: string;
+  hasEverRegistered: boolean;
+  hasAgreedToLatestTerms: boolean;
+  showEmailPreferencesForm: boolean;
+  hasUsername: boolean;
+  blurredAtLeastOnce = false;
+  canReceiveEmailUpdates: boolean;
+  emailPreferencesWarningText: string;
+
+  constructor(
+    private ngbModal: NgbModal,
+    private windowRef: WindowRef,
+    private alertsService: AlertsService,
+    private focusManagerService: FocusManagerService,
+    private loaderService: LoaderService,
+    private signupPageBackendApiService: SignupPageBackendApiService,
+    private siteAnalyticsService: SiteAnalyticsService,
+    private urlService: UrlService
+  ) {}
+
+  ngOnInit(): void {
+    this.loaderService.showLoadingScreen('I18N_SIGNUP_LOADING');
+
+    this.signupPageBackendApiService.fetchSignupPageDataAsync()
+      .then((data) => {
+        this.loaderService.hideLoadingScreen();
+        this.username = data.username;
+        this.hasEverRegistered = data.has_ever_registered;
+        this.hasAgreedToLatestTerms = data.has_agreed_to_latest_terms;
+        this.showEmailPreferencesForm = data.can_send_emails;
+        this.hasUsername = Boolean(this.username);
+        this.focusManagerService.setFocus('usernameInputField');
+      });
+  }
+
+  isFormValid(): boolean {
+    return (
+      this.hasAgreedToLatestTerms &&
+      (this.hasUsername || !this.warningI18nCode)
+    );
+  }
+
+  showLicenseExplanationModal(evt: { target: {innerText: string} }): void {
+    if (evt.target.innerText !== 'here') {
+      return;
+    }
+    let modalRef = this.ngbModal.open(LicenseExplanationModalComponent, {
+      backdrop: true
+    });
+
+    modalRef.result.then(() => {}, () => {
+      // Note to developers:
+      // This callback is triggered when the Cancel button is clicked.
+      // No further action is needed.
+    });
```

Comment:
```suggestion
    // Note to developers:
    // This callback is triggered when the Cancel button is clicked.
    // No further action is needed.
    modalRef.result.then(() => {}, () => {});
```

Reply:
Only the second callback is triggered when the cancel button is clicked. So, it makes more sense to place the comment inside the second callback.

Reply:
But the first callback is empty too, should that be explained too?

Reply:
add comments for both callbacks